python/torch_mlir/extras/fx_importer.py

```diff
@@ -1021,6 +1041,54 @@ def get_node_location(self, node: torch_fx.Node) -> Optional[Location]:
                 return Location.file(filename, line, col=0, context=self._c)
         return Location.unknown(context=self._c)
 
+    def set_symbolic_guards(
+        self, prog: torch.export.ExportedProgram
+    ) -> Dict[str, RangeConstraint]:
+
+        def _sympy_int_to_int(val: sympy.Expr, adjust: str):
+            # Convert simple sympy Integers into concrete int
+            if val == sympy.oo:
+                return math.inf
+            if val == -sympy.oo:
+                return -math.inf
+            if isinstance(val, sympy.Integer):
+                return int(val)
+            # TODO: Remove this adjustment when fractional ranges are removed
+            logging.warning(
+                "Export constraints cannot be non-integer expressions. Found "
+                "type %s, and value %s. We will attempt to %s this value.",
+                type(val),
+                val,
+                adjust,
+            )
+            if adjust == "floor":
+                return math.floor(val)
+            elif adjust == "ceil":
+                return math.ceil(val)
+            else:
+                raise RuntimeError(f"Got invalid adjustment {adjust}")
+
+        contains_symbolic_ints = False
+        for val in prog.range_constraints.values():
+            if (
+                isinstance(val.lower, sympy.Integer)
+                and isinstance(val.upper, sympy.Integer)
+                and not val.is_bool
+            ):
+                contains_symbolic_ints = True
+                break
+        if contains_symbolic_ints:
+            self._symbolic_guards = {
```

Comment:
Can you please add some commentary about the transformation this is doing? I'm a little bit fuzzy on it.

Reply:
Done.